src/user/wikis/Wikis.js

```diff
@@ -0,0 +1,352 @@
+import "./wikis.scss";
+import axios from 'axios';
+import Page from "./Page/Page";
+import Layout from "./Layout/Layout";
+import Editor from "./Editor/Editor";
+import Sidebar from "./Sidebar/Sidebar";
+import { BASE_URL } from "../../actions/baseApi";
+import { getWikis } from "../../actions/wikisAction";
+import { ToastContainer, toast } from "react-toastify";
+
+class Wikis extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      wikis: true,
+      allWikis: [],
+      currentPage: 1,
+      editorMode: false,
+      historyMode: false,
+      viewHistory: false,
+      newPageEditor: false,
+      sidebarEditor: false,
+      spinner: "Loading...",
+    };
+    this.axiosCancel = axios.CancelToken.source();
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.setState({
+      allWikis: nextProps.wikis.wikis,
+      spinner: "",
+    });
+  }
+
+  componentDidMount() {
+    setTimeout(() => {
+      this.props.getWikis();
+    });
+  }
+
+  handleEditorMode = (newPage = false, sidebar = false) => {
+    this.setState({
+      editorMode: true,
+      newPageEditor: !!newPage,
+      sidebarEditor: !!sidebar,
+    });
+  };
+
+  cancelEditor = () => {
+    this.setState({
+      editorMode: false,
+      newPageEditor: false,
+    });
+  };
+
+  setView = async (page, ref) => {
+    let pos = 0;
+    this.state.allWikis.forEach((ele, index) => {
+      if (ele.title === page) pos = index;
+    });
+    if (pos) {
+      this.setState(
+        {
+          spinner: "Switing Page.... ",
+        },
+        async () => {
+          try {
+            let wikis = (
+              await axios.get(`${BASE_URL}/wikis/pages?title=${page}`, {
+                cancelToken: this.axiosCancel.token,
+              })
+            ).data.wikis;
+            wikis.forEach((ele, index) => {
+              if (ele.title === page.title) pos = index;
+            });
+            this.setState({
+              spinner: "",
+              allWikis: wikis,
+              currentPage: pos,
+              viewHistory: false,
+              historyMode: false,
+            });
+          } catch (err) {
+            console.log(err.message);
+          }
+        }
+      );
+    } else {
+      this.handleEditorMode(true);
+    }
+  };
+
+  handleSave = async (page, newPage = false, sidebar = false) => {
+    const { allWikis } = this.state;
+    const endpoint = `${BASE_URL}/wikis/pages`;
+    const data = {
+      title: page.title,
+      content: page.content,
+      comments: page.comments,
+    };
+    const findIndexOfPage = (arr, page) => {
+      let pos = 0;
+      arr.forEach((ele, index) => {
+        if (ele.title === page.title) pos = index;
+      });
+      return pos;
+    };
+    if (!newPage && !sidebar) {
+      this.setState(
+        {
+          spinner: "Saving... ",
+        },
+        async () => {
+          try {
+            const wikis = (
+              await axios.put(endpoint, data, {
```

Comment:
same here, Plz use it as action and dispatch the data from there

Reply:
@Rupeshiya Why? Actions are used to dispatch stuff that is watched by the reducers to make changes to redux store, the changes we intend to make with these async calls are for local state only.

Reply:
But I can tell you the reason because it's not following the code modularity, also it's not all recommended to use either making  the lots of Axios request from the component itself (as I can see in the component itself) because it makes the component  code bigger and not cleaner.
So better to use it as utility functions or as actions.
Hope you got my point.
Let me know if any other doubt.

Reply:
Yeah, I agree with you on that, I felt that too, it makes the code unnecessarily longer. I think we could use utility functions here as you suggested, the other option to use actions would be a little confusing as they are not dispatching anything. I enjoy understanding the reason behind things before implementing them as it leads to better learning. Thanks

Reply:
Done